src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
+  },
```

Comment:
PrimathonAICodeReviewer

Consider adding validation for the `isSuccess` and `statusCode` parameters in the `create` function.
```diff
+     if (isSuccess < 0 || statusCode < 100) {
+         throw new Error('Invalid values for isSuccess or statusCode');
+     }
```


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->